ts/IterableInboxMessageList.tsx

```diff
@@ -30,31 +31,45 @@ const IterableInboxMessageList = ({
 }: MessageListProps) => {
    const [swiping, setSwiping] = useState<boolean>(false)
 
-   const renderMessageCells = (rowViewModels: InboxRowViewModel[]) => {
-      return rowViewModels.map((rowViewModel, index) => {
-         const last = index === rowViewModels.length - 1 ? true : false
-         return (
-            <IterableInboxSwipeableRow
-               key={rowViewModel.inAppMessage.messageId}
-               index={index}
-               last={last}
-               rowViewModel={rowViewModel}
-               customizations={customizations}
-               swipingCheck={(swiping : boolean) => setSwiping(swiping)}
-               messageListItemLayout={messageListItemLayout}
-               deleteRow={(messageId: string) => deleteRow(messageId)}
-               handleMessageSelect={(messageId: string, index: number) => handleMessageSelect(messageId, index)}
-               contentWidth={contentWidth}
-               isPortrait={isPortrait}
-            />
-         )
-      })
-   } 
-
-   return(
-      <ScrollView scrollEnabled={!swiping}>
-         {renderMessageCells(rowViewModels)}
-      </ScrollView>   
+   function renderRowViewModel(rowViewModel: InboxRowViewModel, index: number, last: boolean) {
+      return (
+         <IterableInboxSwipeableRow
+            key = {rowViewModel.inAppMessage.messageId}
+            index = {index}
+            last = {last}
+            rowViewModel = {rowViewModel}
+            customizations = {customizations}
+            swipingCheck = {(swiping: boolean) => setSwiping(swiping)}
+            messageListItemLayout = {messageListItemLayout}
+            deleteRow = {(messageId: string) => deleteRow(messageId)}
+            handleMessageSelect = {(messageId: string, index: number) => handleMessageSelect(messageId, index)}
+            contentWidth = {contentWidth}
+            isPortrait = {isPortrait}
+         />
+      )
+   }
+
+   const inboxSessionViewabilityConfig: ViewabilityConfig = {
+      minimumViewTime: 0.5,
+      itemVisiblePercentThreshold: 50,
+      waitForInteraction: true
+   }
+
+   const inboxSessionItemsChanged = useCallback((
+      (info: {viewableItems: Array<ViewToken>, changed: Array<ViewToken>}) => {
+         
+      }
+   ), [])
+
+   return (
+      <FlatList
+         scrollEnabled = {!swiping}
+         data = {rowViewModels}
+         renderItem = {({item, index}: {item: InboxRowViewModel, index: number }) => renderRowViewModel(item, index, index === rowViewModels.length - 1)}
+         keyExtractor = {(item: InboxRowViewModel) => item.inAppMessage.messageId}
+         viewabilityConfig = {inboxSessionViewabilityConfig}
+         onViewableItemsChanged = {inboxSessionItemsChanged}
+      />
```

Comment:
I like the refactor. I think render Item should be written as follows:
renderItem = {(item: InboxRowViewModel, index: number) => renderRowViewModel(item, index, index === rowViewModels.length - 1)}

Reply:
I see this is the required way to add a type for item and index